OpenRA.Mods.Common/Traits/BlocksProjectiles.cs

```diff
@@ -58,7 +58,12 @@ public static bool AnyBlockingActorsBetween(World world, WPos start, WPos end, W
 				if (!blockers.Any())
 					continue;
 
-				var hitPos = WorldExtensions.MinimumPointLineProjection(start, end, a.CenterPosition);
+				// FindActorsOnLine only finds actors with health trait, so this actor is guaranteed to have it
+				var healthInfo = a.Info.TraitInfo<HealthInfo>();
+				var vector = (a.CenterPosition - source) * healthInfo.Shape.InnerRadius.Length / (a.CenterPosition - source).Length;
```

Comment:
Likewise for `InnerRadius`.  You care about the direction rather than the length, so you shouldn't need to scale this at all.